### PR TITLE
fix: limit length of labels in legend panel (#2329)

### PR DESCRIPTION
**Pull request by luelista**
# fix: limit length of labels in legend panel

Issue: #2329

## Description

Cut off overlong labels in the legend panel, so it doesn't take the width of the whole screen. The full label is visible as a tooltip.

## Screenshots or gifs

grafik

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/)
 ### Fixed 🐞
 
 -   Update slider of area metric options correctly on changes of related input field [#2787](https://github.com/MaibornWolff/codecharta/pull/2787)
+-   Limit length of labels in legend panel #2329
```
Comment (shaman-apprentice) [inline CHANGELOG.md:13]: This GitHub linking doesn't work with the changes information we display to the user within CodeCharta, which are generated by this changelog. Although we might add this parsing to the generation script.

We link the PR and not the issue. Feel very free to address that. I find it more natural to link the issue as well :)

```suggestion
-   Limit length of labels in legend panel [#2804](https://github.com/MaibornWolff/codecharta/pull/2804)
```

File: visualization/app/codeCharta/ui/legendPanel/legendPanel.component.scss [hunk 2]
```diff
@@ -70,19 +72,25 @@ cc-legend-panel {
...
 		.bold {
 			font-weight: 600;
 		}
+
+		.cc-labelled-color-picker-container > .cc-labelled-color-picker-label {
+			white-space: nowrap;
+			text-overflow: ellipsis;
+			overflow: hidden;
+
+			/* This tricks the browser into cutting off the beginning instead of the end of the label. However,
```
Comment (shaman-apprentice) [inline visualization/app/codeCharta/ui/legendPanel/legendPanel.component.scss:86]: I am not too sure about the comments. They describe standard css. Maybe rename the css selector to `cc-cut-text-overflow-at-beginning` instead? On the other side there is no harm by the comments. I am fine with leaving it as it is as well.

**Review by shaman-apprentice — APPROVED** [review]
Early approve. Just the changelog entry and the optional comment about the comment (commentception :P).

Random information: I like to use [GitHub's closing words](https://docs.devart.com/studio-for-sql-server/source-controlling-databases/associating-commits-with-github-issues.html) in PR description.

File: visualization/app/codeCharta/ui/legendPanel/legendPanel.component.scss (hunk 2 above)
Comment (luelista) [reply visualization/app/codeCharta/ui/legendPanel/legendPanel.component.scss:86]: IMO this is still a misuse, or at least exploiting an edge case of standard CSS. The `direction: rtl;` property is not intended to switch text-overflow cutoff, but for right-to-left languages, which this is not. So as this a hack with some weird side effects, I want to make sure there is some warning at least.

**Pull request merged**